### PR TITLE
chore: adjust CI to respect nestjs 10 dropping EOL node versions

### LINKED ISSUE #682: Support for @nestjs/common 10.0.0 in Peer Dependencies
Hello there! Thank you for your contribution to this library.

I have opened this issue to inquire about the possibility of adding support for NestJS version 10.0.0. Currently, the library's peer dependency only supports up to version 9.0.0. Considering that NestJS has released a new major version, it would be great to have compatibility with the latest version.

Is there any plan or timeline to add support for NestJS 10.0.0? I believe many users would greatly benefit from this addition.

Thank you!

### PR DESCRIPTION
**Pull request by adrianmxb**
In the [NestJS 10 Release](https://github.com/nestjs/nest/releases/tag/v10.0.0) it was announced that the supported node.js versions are >= 16. 
Adjusted the CI, so it runs again and the tests run properly only on supported versions. :)

Closes #682

### CODE REVIEW

**Comment by gremo** [comment]
Thank you a lot @adrianmxb, this is really something needed! can you please clarify what matrix elements (jobs) we are dropping?

File: .github/workflows/test-on-push.yml [hunk 1]
```diff
@@ -19,6 +19,16 @@ jobs:
           - { node-version: '10', nest-version: '6' }
           - { node-version: '10', nest-version: '7' }
           - { node-version: '10', nest-version: '8' }
+          - { node-version: '12', nest-version: '5' }
```
Comment (gremo) [inline .github/workflows/test-on-push.yml:22]: Why moving node 12 from matrix above to include?

File: .github/workflows/test-on-push.yml [hunk 2]
```diff
@@ -19,6 +19,16 @@ jobs:
...
+          - { node-version: '12', nest-version: '6' }
+          - { node-version: '12', nest-version: '7' }
+          - { node-version: '12', nest-version: '8' }
+          - { node-version: '12', nest-version: '9' }
+          - { node-version: '14', nest-version: '5' }
```
Comment (gremo) [inline .github/workflows/test-on-push.yml:27]: Same as above...

**Comment by adrianmxb** [comment]
Thanks for acting so fast, really appreciated! :)

**Pull request merged**